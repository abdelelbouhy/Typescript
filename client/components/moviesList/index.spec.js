import React from 'react';
import {mount} from 'enzyme';
import MoviesList from './index.tsx';
import reducers from '../../../redux/reducers';
import {combineReducers, createStore} from 'redux';
import {routerReducer} from 'react-router-redux';
import {updateList} from '../../../redux/actions';

let store = createStore(
	combineReducers({
		...reducers,
		routing: routerReducer
	})
);

store.dispatch(updateList([{Title: 'Test Title1'}, {Title: 'Test Title2'}]));

describe('Movies List Component', () => {
	it('should render List of movies', () => {
		const wrapper = mount(<MoviesList store={store}/>);

		expect(wrapper).toMatchSnapshot();
	});
});