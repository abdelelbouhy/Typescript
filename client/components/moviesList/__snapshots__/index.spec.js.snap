// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies List Component should render List of movies 1`] = `
<Connect(MoviesList)
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <MoviesList
    moviesList={
      Object {
        "data": Array [
          Object {
            "Title": "Test Title1",
          },
          Object {
            "Title": "Test Title2",
          },
        ],
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
    updateMoviesList={[Function]}
  >
    <div
      className="movies_list-wrapper"
    >
      <h1
        className="movies_list-wrapper-heading"
      >
        Search Movies By
      </h1>
      <form
        className="movies_list-wrapper-search_form"
      >
        <div
          className="movies_list-wrapper-search_form-field"
          key="title"
        >
          <label
            className="movies_list-wrapper-search_form-label"
            htmlFor="movies_list-wrapper-search_form-search"
          >
            Title
            :
          </label>
          <input
            className="movies_list-wrapper-search_form-search_input"
            id="title"
            name="s"
            onChange={[Function]}
            placeholder="Please enter a movie title"
            type="text"
            value=""
          />
        </div>
      </form>
      <div
        className="movies_list-wrapper-movies_list"
      >
        <div
          className="movies_list-wrapper-movies_list-movie"
          key="undefined_0"
        >
          <Link
            className="movies_list-wrapper-movies_list-movie-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/moviesItem/Test Title1"
          >
            <a
              className="movies_list-wrapper-movies_list-movie-link"
              onClick={[Function]}
              style={Object {}}
            >
              Test Title1
            </a>
          </Link>
        </div>
        <div
          className="movies_list-wrapper-movies_list-movie"
          key="undefined_1"
        >
          <Link
            className="movies_list-wrapper-movies_list-movie-link"
            onlyActiveOnIndex={false}
            style={Object {}}
            to="/moviesItem/Test Title2"
          >
            <a
              className="movies_list-wrapper-movies_list-movie-link"
              onClick={[Function]}
              style={Object {}}
            >
              Test Title2
            </a>
          </Link>
        </div>
      </div>
      <div
        className="clear-fix"
      />
      <div
        className="movies_list-wrapper-add_search_criteria"
        onClick={[Function]}
      >
        Add Search Criteria
      </div>
      <div
        className="clear-fix"
      />
    </div>
  </MoviesList>
</Connect(MoviesList)>
`;
